storage-proofs/src/stacked/proof.rs

```diff
@@ -386,7 +402,26 @@ impl<'a, H: 'static + Hasher, G: 'static + Hasher> StackedDrg<'a, H, G> {
             }
 
             // Write the result to disk to avoid keeping it in memory all the time.
-            labels.push(DiskStore::new_from_slice(layer_size, &layer_labels)?);
+            let layer_config = StoreConfig::from_config(
+                &config,
+                format!("replica-{:?}-layer-{}", replica_id, layer),
+                Some(layer_size),
+            );
+
+            // Construct and persist the layer data.
+            let _: DiskStore<H::Domain> = DiskStore::new_from_slice_with_config(
```

Comment:
this method should just return a `LabelCache` so you don't need to recreate the stores when using it